### ISSUE TITLE
Unavailable to build with go mod

### ISSUE THREAD

**Issue by spg83** [chaos-mesh/chaos-mesh#3715]
## Feature Request

**Is your feature request related to a problem? Please describe:**

It's unavailable to build with `v2` with `go mod` toolchain, this is the `go get` errors

```
go get github.com/chaos-mesh/chaos-mesh@v2.4.1
go: github.com/chaos-mesh/chaos-mesh@v2.4.1: invalid version: module contains a go.mod file, so module path must match major version ("github.com/chaos-mesh/chaos-mesh/v2")
```

**Describe the feature you'd like:**

This repository should follow `go mod`'s major revision policy, that is, the module should be declared as `github.com/chaos-mesh/chaos-mesh/v2` at `github.com/chaos-mesh/chaos-mesh/go.mod`.

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

**Comment by g1eny0ung** [comment]
@STRRL Should updating to `github.com/chaos-mesh/chaos-mesh/v2` suit our codebase?

**Comment by STRRL** [comment]
also cc @YangKeao @cwen0, how do you think about it ?

**Comment by spg83** [comment]
@STRRL Yes, there are bundles of thing have to be fixed to publish `v2`. 
Currently I have to use `git submodule` and `go mod replace` to work around, however, I prefer to go back to `go mod sematics`.

**Comment by STRRL** [comment]
> @STRRL Yes, there are bundles of thing have to be fixed to publish `v2`. Currently I have to use `git submodule` and `go mod replace` to work around, however, I prefer to go back to `go mod sematics`.

It might take a long time to change to the server, before that, you could use the pseudo version with `go get github.com/chaos-mesh/chaos-mesh@bb212d4`

**Comment by spg83** [comment]
> It might take a long time to change to the semver version, before that, you could use the pseudo version with go get github.com/chaos-mesh/chaos-mesh@bb212d4

Trying to use this pseudo version at v2.4.1(`bb212d4`), then stuck in `chaos-mesh/api` problem. 
It's seems that module `chaos-mesh` use a dependency for `chaos-mesh/api` at `v0.0.0`, but it doesn't exist.

```
go get github.com/chaos-mesh/chaos-mesh@bb212d4b
go mod tidy
...
go: downloading github.com/chaos-mesh/chaos-mesh/api v0.0.0
...
github.com/chaos-mesh/chaos-mesh/api/v1alpha1: reading github.com/chaos-mesh/chaos-mesh/api/api/go.mod at revision api/v0.0.0: unknown revision api/v0.0.0
...
```

I can't fix it although use pseudo version for chaos-mesh/api`, because go module will remove `chaos-mesh` module 
```
go get github.com/chaos-mesh/chaos-mesh/api@bb212d4b
go: removed github.com/chaos-mesh/chaos-mesh v0.9.1-0.20220927131145-bb212d4be84c
go: downgraded github.com/chaos-mesh/chaos-mesh/api v0.0.0 => v0.0.0-20220927131145-bb212d4be84c
```

**Comment by STRRL** [comment]
You should use `go replace` also with `github.com/chaos-mesh/chaos-mesh/api`.

>  go module will remove chaos-mesh module

I do not get your point, go mod would only remove the dependency if there is not code using it.

**Comment by spg83** [comment]
> I do not get your point, go mod would only remove the dependency if there is not code using it.

I don't know why the go mod would remove `chaos-mesh/chaos-mesh`, it's a bit confused, but this is another issue.

I am not sure if it's the original problem: [go.mod#L18](https://github.com/chaos-mesh/chaos-mesh/blob/release-2.4/go.mod#L18)

**Comment by STRRL** [comment]
> I don't know why the go mod would remove chaos-mesh/chaos-mesh, it's a bit confused, but this is another issue.

go would delete the dependency entry if there are no codes in `.go` file use it.
